src/options.rs

```diff
@@ -105,9 +91,14 @@ impl Options {
             algorithm: Algorithm::from_str(matches.value_of("algorithm").unwrap()).unwrap(),
             verify: verify,
             depth: if matches.is_present("recursive") {
-                DepthSetting::Infinite
+                None
             } else {
-                DepthSetting::from_str(matches.value_of("depth").unwrap_or("0")).unwrap()
+                let i = matches.value_of("depth").map(|s| s.parse::<isize>().unwrap()).unwrap_or(1);
+                if i <= 0 {
```

Comment:
Note that depth is shifted from what it was: Before, 0 meant only the entries in the directory, 1 would be those, and their children. It is now <= 0 for infinite recursion, 1 means only the direct children of the given directory. This is more consistent with my intuition (depth=2 means two levels) and is more consistent with other utils (e.g. find --max-depth)